osu.Game/Screens/Edit/Compose/Components/EditorSelectionHandler.cs

```diff
@@ -39,6 +39,23 @@ private void load()
             SelectedItems.CollectionChanged += (_, _) => Scheduler.AddOnce(UpdateTernaryStates);
         }
 
+        protected override void BeginChange()
+        {
+            // If the editor beatmap has a change handler then this will automatically call BeginChange() on it.
+            EditorBeatmap.BeginChange();
+        }
+
+        protected override void EndChange()
+        {
+            foreach (var hitObject in SelectedItems)
+            {
+                EditorBeatmap.Update(hitObject);
+            }
```

Comment:
Same with this thing, I'm not sure where this materialised from (and its copy in `OsuSelectionRotationHandler`, which is especially painful for me personally, specifically). 

Am I thinking correctly that this is because the base `SelectionHandler` is attempting to resolve `IEditorChangeHandler`? If so, can we try caching the `EditorBeatmap` itself as `IEditorChangeHandler` or whatever in hopes of getting the old behaviour back? To me that's still better than current `master`, because even though you would be able to resolve either via `EditorBeatmap` or `IEditorChangeHandler`, you would at least be resolving _the same object in both cases_, rather than two different ones as currently.

Reply:
This updating of hitobjects on the `EditorBeatmap` is completely unrelated.
Previously, the operations on selected hit objects by the `EditorSelectionHandler` never called `Update` for the hit objects it updated, resulting in the `EditorBeatmap` not knowing which hit objects got updated. Current editor flow requires that the operations manually notify the `EditorBeatmap` for any hit objects that it updates.

Reply:
Ok, so this is fixing a bug.

I can live with this, so long as there is only one copy of it present. We can probably make that happen by making `OsuSelectionRotationHandler` accept `OsuSelectionHandler` in ctor rather than interfacing with `EditorBeatmap` directly. Mind making that change?

Reply:
Seems kinda weird to make it depends on `OsuSelectionHandler`. It seemed to me that the `OsuSelectionRotationHandler` was built to be able to work without the selection handler.

Reply:
That wasn't explicitly a goal when I made it. The primary goal was logic reuse. The primary version I wrote actually used `SelectionHandler`, I believe, but I backed out of it in order to not introduce too many generics.

My point is that the fact that this has to be explicitly handled in multiple places is ugly to me and I'd prefer to not do that if feasible.

Reply:
How will adding the `OsuSelectionHandler` to the `OsuSelectionRotationHandler` help reduce this code duplication? The `BeginChange` and `EndChange` methods are protected. Should I make them public? This would make the change handling even more convoluted with some things calling `SelectionHandler.BeginChange()` and other things calling `EditorBeatmap.BeginChange`. That seems to defeat the whole purpose of this PR. 

I intend to remove the need for `EditorBeatmap.Update` in the future anyways by implementing a command pattern, so I think it's fine to have it like this for now.